torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -1937,6 +1937,25 @@ at::Tensor XLANativeFunctions::mul(const at::Tensor& self,
                     });
 }
 
+at::Tensor XLANativeFunctions::multinomial(
+    const at::Tensor& self, int64_t num_samples, bool replacement,
+    c10::optional<at::Generator> generator) {
+  TORCH_LAZY_FN_COUNTER("xla::");
+  // Fallback when sampling is not replaced because it is challenging to
+  // parallelize.
+  if ((generator.has_value() && generator->defined()) ||
+      (!replacement && num_samples > 1)) {
```

Comment:
Let's be more explicit by saying `num_samples != 1`

Reply:
Also do we need any error handling when num_samples < 1?

Reply:
Good catch!